### ISSUE TITLE
bounds properties

### ISSUE THREAD

**Issue by pnorman** [osmlab/openstreetmap-json-schema#9]
From Roland Olbricht (@drolbr)

> Similarly, what is the exact meaning of the "bounds" field?
> 
> If it should be the parameters the user has originally requested then it
> would make most sense to drop the bounds when unifying two JSON sets,
> and subsequently to allow at most one bounding box or none.
> 
> If it should somehow relate to the payload, like the assertion that all
> nodes in the existing database are inside, then of course it would make
> sense to be able to express this assertion in more than one bounding
> box.
> 
> If by contrast the bounding box gives the assertion that all elements
> are inside it (but not necessarily complete), then again multiple
> bounding boxes may not make sense because in a unification of two files
> the bounding box would rather expand to the outer bounding box of all
> files.
> 
> So the important thing here would be to clarify the meaning of the
> bounding box. When I was asked to include a bounding box, I was always
> asked for to return the user requested parameters, which still doesn't
> make much sense to me but seems to be the most used thing.

Just as a reminder, the spec currently calls for

``` javascript
"bounds": {
    // If the bounds are present, all four keys are REQUIRED and are
    // floating point numbers that form a bounding box.
    "minlat": -90.0,
    "minlon": -180.0,
    "maxlat": 90.0,
    "maxlon": 180.0
}
```

The bounds field is somewhat odd as it currently only ever returns the bbox parameter to the map call, and this information is part of the URL requested by the API

I see three potential uses for it
1. Returning the information for clients which rely on its presence instead of knowing what they requested. JOSM is like this, I do not believe it uses its request to derive bounds in absence of a <bounds/> element
   
   There are occasions where a client cannot know what they requested, for example where a file is saved and then opened because the query parameters are normally lost doing this.
   
   The current bounds syntax can be used to meet this requirement.
2. An API implementation where `map?bbox=$BOUNDS` where the return is guaranteed to include `$BOUNDS` but may include more. This might make sense with a tiled backend where you want to return an entire tile when only part of it was actually requested.
   
   The current bounds syntax can be used to meet this requirement.
3. Requesting multiple bboxes or merging multiple requests. JOSM does this.
   
   The current bounds syntax **cannot** be used to meet this requirement.
   
   What JOSM does (multiple `<bounds/>` is best represented in json as
   
   ``` javascript
   "bounds": [
       {
           "minlat": 49.17,
           "minlon": -122.96,
           "maxlat": 49.21,
           "maxlon": -122.9
       },
       {
           "minlat": 49.2,
           "minlon": -122.95,
           "maxlat": 49.24,
           "maxlon": -122.87
       }
   }
   ```

**Comment by pnorman** [comment]
Personally, I prefer the JOSM way.

**Comment by jfirebaugh** [comment]
Is there anything JSON-specific to these considerations, or is all the ambiguity already existent within the XML format? If the latter, I don't see a need to block work towards JSON API support on resolving the ambiguity here.

**Comment by joto** [comment]
Yes, all of this isn't JSON-specific. Because of the current mess with the bounds in other formats, I suggest we define the bounds as a "hint" only that might or might not give you the bbox of the actual data in the file. Users are not supposed to rely on it. This is basically current use anyway.

**Comment by pnorman** [comment]
> or is all the ambiguity already existent within the XML format?

The XML format allows multiple <bounds/> elements, e.g. `<bounds minlat=... /><bounds minlat=...2 />` with appropriate attributes is valid XML, but the equivalent JSON is not valid.
